src/plugins/condition/StyleRuleManager.js

```diff
@@ -0,0 +1,77 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2020, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+import * as EventEmitter from 'eventemitter3';
+
+export default class StyleRuleManager extends EventEmitter {
+    constructor(domainObject, openmct) {
+        super();
+        this.domainObject = domainObject;
+        this.openmct = openmct;
+        if (this.domainObject.configuration && this.domainObject.configuration.conditionalStyle) {
+            this.conditionSetIdentfier = this.domainObject.configuration.conditionalStyle.conditionSetIdentifier;
+            this.defaultStyle = this.domainObject.configuration.conditionalStyle.defaultStyle;
+            this.conditionalStyles = this.domainObject.configuration.conditionalStyle.styles || [];
+            this.initialize();
+        }
+    }
+
+    initialize() {
+        this.openmct.objects.get(this.conditionSetIdentfier).then((obj) => {
+            this.stopProvidingTelemetry = this.openmct.telemetry.subscribe(obj, output => this.handleConditionSetResultUpdated(output));
+        });
```

Comment:
It will also be necessary to request historical telemetry here to get the "latest" state value. eg.

``` JavaScript
this.openmct.telemetry.request(conditionSet, {strategy: 'latest', size: 1}).then(...)
```

Reply:
What was the use-case for this again?

Reply:
To determine the initial state of the condition set. Let's say you have a telemetry point that only produces new data every 30 seconds. When you load the screen, the ConditionSet could sit and wait 30 seconds to receive realtime data to determine its state, or it could use historical data to determine the current state of the system. We are only interested in the *most recent* value returned though, so use the last value in the returned array.

Reply:
I understand now. Thanks. I'll do that in a separate issue - #2723

Reply:
@shefali sounds good.